flocker/node/agents/gce.py

```diff
@@ -754,3 +815,27 @@ def _stop_node(self, node_id):
             timeout_sec=5*60,
             instance=node_id
         )
+
+
+def gce_from_configuration(cluster_id, project=None, zone=None,
+                           credentials=None):
+    """
+    Build a ``GCEBlockDeviceAPI`` instance using data from configuration
+
+    :param UUID cluster_id: The unique identifier of the cluster with which to
+        associate the resulting object.  It will only manipulate volumes
+        belonging to this cluster.
+    :param str project: The GCE project for the cluster
+    :param str zone: The GCE zone the cluster will be located in
```

Comment:
Sorry, i was just being lazy, they actually are unicode (read in from a yaml file) certainly get_machine_project is documented as returning a unicode.